homeassistant/components/here_weather/utils.py

```diff
@@ -0,0 +1,55 @@
+"""Utility functions for here_weather."""
+from __future__ import annotations
+
+from homeassistant.const import (
+    CONF_UNIT_SYSTEM_METRIC,
+    LENGTH_CENTIMETERS,
+    LENGTH_INCHES,
+    LENGTH_KILOMETERS,
+    LENGTH_MILES,
+    PRESSURE_INHG,
+    PRESSURE_MBAR,
+    SPEED_KILOMETERS_PER_HOUR,
+    SPEED_MILES_PER_HOUR,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+
+
+def convert_unit_of_measurement_if_needed(
+    unit_system, unit_of_measurement: str | None
+) -> str | None:
+    """Convert the unit of measurement to imperial if configured."""
+    if unit_system != CONF_UNIT_SYSTEM_METRIC:
+        if unit_of_measurement == TEMP_CELSIUS:
+            unit_of_measurement = TEMP_FAHRENHEIT
+        elif unit_of_measurement == LENGTH_CENTIMETERS:
+            unit_of_measurement = LENGTH_INCHES
+        elif unit_of_measurement == SPEED_KILOMETERS_PER_HOUR:
+            unit_of_measurement = SPEED_MILES_PER_HOUR
+        elif unit_of_measurement == PRESSURE_MBAR:
+            unit_of_measurement = PRESSURE_INHG
+        elif unit_of_measurement == LENGTH_KILOMETERS:
+            unit_of_measurement = LENGTH_MILES
+    return unit_of_measurement
+
+
+def get_attribute_from_here_data(
+    here_data: list, attribute_name: str, sensor_number: int = 0
+) -> str | None:
+    """Extract and convert data from HERE response or None if not found."""
+    if here_data is None:
```

Comment:
This check doesn't match the typing in the signature. Generally it's better to check the type of a parameter before calling the next function to have cleaner typing and code.

Reply:
Implemented in https://github.com/home-assistant/core/pull/28910/commits/5ab2e3c06356de7d32b02a171f535de52f7ea226

Reply:
And fixed in https://github.com/home-assistant/core/pull/28910/commits/15e174d00a013cf285c8f23eb018e079d3ca5e09